packages/calypso-e2e/src/browser-manager.ts

```diff
@@ -114,11 +114,31 @@ export async function close(): Promise< void > {
 }
 
 /**
- * Given a page, this will clear the cookies for the context to which the page belongs.
+ * Given a page, this will clear all cookies, local storage and reset permissions for the
+ * Browser Context to which the page belongs.
  *
  * @param {Page} page Object representing a page launched by Playwright.
  * @returns {Promise<void>} No return value.
  */
 export async function clearCookies( page: Page ): Promise< void > {
-	await page.context().clearCookies();
+	// Save references to the BrowserContext and the current URL the page is on.
+	const browserContext = page.context();
+	const currentURL = page.url();
+
+	// Navigate to the WordPress.com base URL.
+	await page.goto( config.get( 'calypsoBaseURL' ) );
+	// Clear the GDPR banner cookie.
+	await page.evaluate( 'window.document.cookie = "sensitive_pixel_option=no;";' );
+	// Clear additional cookies.
+	await page.evaluate(
+		'window.document.cookie = "sensitive_pixel_option=no;domain=.wordpress.com;SameSite=None;Secure"'
+	);
+	// Clear cookies to remote login.
```

Comment:
Could you explain a little more what's going on in these lines?

From what I can tell, the goal is to wipe cookies (which is what we do in the end with `browserContext.clearCookies()`). It looks like the goal in these previous steps is mostly to wipe `localstorage` on `r-login.wordpress.com`. 

Is there any reason we can't trim this down to:
```typescript
	await page.goto( 'https://r-login.wordpress.com/' );
	// Clear local storage.
	await page.evaluate( 'localStorage.clear();' );
	// Lastly, clear the cookies using built-in method.
	await browserContext.clearCookies();
	// Previous steps navigated page away from target page. Return page to the original URL.
	await page.goto( currentURL );
```

Reply:
I copied this behavior from the existing Selenium suite. This behavior can be found [here](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/lib/driver-manager.js#L353-L357), [here](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/lib/driver-manager.js#L304-L313) and [here](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/lib/driver-manager.js#L319-L343).

I think it may be worth trying to see if what you suggested suffices though.

Reply:
You were correct!